tao_compiler/mlir/disc/transforms/lhlo_legalize_roots_to_loops.cc

```diff
@@ -3660,6 +3660,190 @@ LogicalResult lowerWithScheduleLargeConcatCPU(
   return success();
 }
 
+// SparseReshape implementation:
+//
+// Loop over input
+// for (i = 0; i < input_indices.shape(0), ++i) {
+//   // multidim index to linear index
+//   linear_index = ...;
+//
+//   // linear index to multidim index with new shape
+//   multidim_index = ...;
+// }
+// TODO(lanbo.llb): Support SparseReshapeOp on GPU
+LogicalResult lowerWithScheduleSparseReshapeOpCPU(
```

Comment:
Add an end2end test for SparseReshape?

Reply:
e2e test is in tao_compiler/mlir/disc/tests/tensorflow_ops/sparse_reshape.cc